vizier/backend/src/info/vizierdb/commands/plot/ScatterPlot.scala

```diff
@@ -0,0 +1,335 @@
+
+object ScatterPlot extends Command
+{
+  val PARAM_SERIES = "series" 
+  val PARAM_DATASET = "dataset"
+  val PARAM_X = "xcol"
+  val PARAM_Y = "ycol"
+  val PARAM_FILTER = "filter"
+  val PARAM_COLOR = "color"
+  val PARAM_LABEL = "label"
+  val PARAM_ARTIFACT = "artifact"
+
+  val MAX_RECORDS = 10000
+
+  override def name: String = "Scatter Plot"
+
+  override def parameters: Seq[Parameter] = Seq(
+    ListParameter(id = PARAM_SERIES, name = "Lines", components = Seq(
+      DatasetParameter(id = PARAM_DATASET, name = "Dataset"),
+      ColIdParameter(id = PARAM_X, name = "X-axis"),
+      ColIdParameter(id = PARAM_Y, name = "Y-axis"),
+      StringParameter(id = PARAM_LABEL, name = "Label", required = false),
+      StringParameter(id = PARAM_FILTER, name = "Filter", required = false),
+      StringParameter(id = PARAM_COLOR, name = "Color", required = false),
+    )),
+    StringParameter(id = PARAM_ARTIFACT, name = "Output Artifact (blank to show only)", required = false)
+  )
+  override def title(arguments: Arguments): String = 
+    "PLOT "+arguments.getList(PARAM_SERIES).map { series =>
+      series.pretty(PARAM_DATASET)
+    }.toSet.mkString(", ")
+
+  override def format(arguments: Arguments): String = 
+    "PLOT "+arguments.getList(PARAM_SERIES).map { series =>
+      f"${series.pretty(PARAM_DATASET)}.{${series.pretty(PARAM_X)}, ${series.pretty(PARAM_Y)}}${series.getOpt[String](PARAM_LABEL).map { " AS " + _ }.getOrElse("")}"
+    }.mkString("\n     ")
+
+  override def process(arguments: Arguments, context: ExecutionContext): Unit = 
+  {
+    //Variables to store axes titles
+    var xTitle = ""
+    var yTitle = ""
+
+    //Variables to hold the x and y values
+    var xValues = scala.collection.mutable.ListBuffer[Double]()
+    var yValues = scala.collection.mutable.ListBuffer[Double]()
+
+    //Variables to hold all of the dataset names, y columns, x columns, and combinations of them along with boolean values that can be used to identify if those naming conventions create all unique names
+    var datasetNames = scala.collection.mutable.Map[String, Int]()
+    var datasetUnique = true
+    var yColumns = scala.collection.mutable.Map[String, Int]()
+    var yUnique = true
+    var xColumns = scala.collection.mutable.Map[String, Int]()
+    var xUnique = true
+    var datasetAndY = scala.collection.mutable.Map[String, Int]()
+    var datasetAndYUnique = true
+    var datasetAndX = scala.collection.mutable.Map[String, Int]()
+    var datasetAndXUnique = true
+
+    // Figure out if we are being asked to emit a named artifact
+    // Store the result in an option-type
+    val artifactName = arguments.getOpt[String](PARAM_ARTIFACT)
+                                .flatMap { case "" => None 
+                                           case x => Some(x) }
+
+    
+
+    //meaningless series object that's just here to iterate through all of the dataset names, x axes, and y axes so we can check for uniqueness
+    var tempSeries = 
```

Comment:
In the following: 
```
var tempSeries = arguments.getList(PARAM_SERIES).map { series => ... }
```

The variable `tempSeries` is never actually used.  Please replace this with a for loop:
```
for(series <- arguments.getList(PARAM_SERIES)) 
{
   ...
}
```

Reply:
Alternatively, I would propose the following way of framing the problem: Decouple the information extraction and the computation steps:

First, Extract the information: You're interested in the set of dataset names, column names, etc...  First, generate a representation of this (I would suggest a sequence of tuples: `Seq[(String, String, String)]` ).  Take a look at the docs for [Seq.map](https://www.scala-lang.org/api/2.13.8/scala/collection/Seq.html#map[B](f:A=%3EB):CC[B]).  You can do this in ~4 lines.  

Once you have that, you can express the constraints quite a bit more efficiently: For example:
```
val nameComponents:Seq[(String, String, String)] = arguments.getList(PARAM_SERIES).map {series => ... }

val numberOfUniqueDatasetNames = 
  nameComponents.map { c => c._1 }.toSet.size
if(numberOfUniqueDatasetNames < nameComponents.size){ ... }
```

This approach has the benefit of avoiding mutable state entirely... and mutable state is a huge source of potential bugs.